torch/_dynamo/output_graph.py

```diff
@@ -1053,6 +1059,15 @@ def call_user_compiler(self, gm: fx.GraphModule) -> CompiledFn:
             )
             unimplemented_with_warning(e, self.root_tx.f_code, msg)
         except Exception as e:
+            if (
+                isinstance(e, (UnbackedSymIntError, GuardOnDataDependentSymNode))
+                or (
+                    isinstance(e, TypeError)
+                    and e.args[0] == "Cannot convert symbols to int"
```

Comment:
still string matching 🤔

Reply:
This is used to make one corner case in testing compiled autograd work well, it seems compiled autograd produced ```nonzero``` call and this pre_grad pass can't capture it. I'm not sure if it's common that autograd added these calls.  For all other cases, we don't need it.